include/aspect/simulator.h

```diff
@@ -839,6 +839,15 @@ namespace aspect
                                             LinearAlgebra::Vector &vec);
 
       /**
+       * Fill @p constraints with a zero Dirichlet constraints for the first
+       * velocity unknown in the domain for each velocity component, which is
+       * later being processed for translational or linear momentum removal.
+       * While this is not the correct value for the velocity component, it ensures
+       * that the linear system is invertible.
+       */
+      void setup_nullspace_constraints(ConstraintMatrix &constraints);
```

Comment:
the function really only does translational nullspaces. do you want to rename the function maybe, or is the intent to deal with all possible null spaces but (for now) simply not do anything about rotational modes? In the latter case, maybe a sentence in the documentation, or a comment in the code would help to understand what is supposed to happen in those cases.